framework/src/outputs/Checkpoint.C

```diff
@@ -47,6 +54,8 @@ Checkpoint::validParams()
 
 Checkpoint::Checkpoint(const InputParameters & parameters)
   : FileOutput(parameters),
+    _should_output(getParam<bool>("should_output")),
```

Comment:
You should be able to initialize this Boolean based upon the state of the other one. The other one should be marked `const`.